lib/spack/spack/test/cmd/checksum.py

```diff
@@ -272,3 +272,12 @@ def test_checksum_verification_fails(install_mockery, capsys):
     assert out.count("Correct") == 0
     assert "No previous checksum" in out
     assert "Invalid checksum" in out
+
+
+def test_checksum_manual_download_fails(mock_packages, monkeypatch):
+    """Confirm that checksumming a manually downloadable package fails."""
+    pkg_cls = spack.repo.PATH.get_pkg_class("zlib")
+    versions = [str(v) for v in pkg_cls.versions]
+    monkeypatch.setattr(spack.package_base.PackageBase, "manual_download", True)
+    with pytest.raises(spack.main.SpackCommandError):
+        spack_checksum("zlib", *versions)
```

Comment:
The PR in general looks good, but we might want to make sure our error output specifically talks about zlib being a manual download (otherwise all sorts of unrelated issues could cause this invocation to fail).

Reply:
Do you mean this test should be explicitly checking the error output?

Because the error message is the manual download instructions that, by default (so if not overridden) provide the name of the package, says manual download required, and includes the homepage.

Reply:
FWIW I started to do that but thought someone might complain that the output didn't really need to be checked by the unit test.